manager/apis/app/v1alpha1/fybrikmodule_types.go

```diff
@@ -94,23 +109,50 @@ type ModuleAPI struct {
 	Endpoint EndpointSpec `json:"endpoint"`
 }
 
+// PluginMechanism indicates indicates the technology used for the module and the plugin to interact
+// The values supported should come from the module taxonomy
+// Ex: vault, wasm....
+type PluginMechanism string
+
+type Plugin struct {
+	// PluginType indicates the technology used for the module and the plugin to interact
+	// Examples of such mechanisms are vault plugins, wasm, etc
+	// +required
+	PluginType PluginMechanism `json:"pluginType"`
+
+	// DataFormat indicates the format of data the plugin knows how to process
+	DataFormat string `json:"dataFormat"`
+}
```

Comment:
How does this structure indicate what actions can be loaded as plugins, where are the libraries found (images)?

Reply:
@shlomitk1 This structure indicates that the module supports these types of plugins (ex: vault plugins). It doesn't indicate the specific plugin modules that it supports. Rather, any plugin module of this type/format can be used. Think of it like an electrical wall socket. It can support any "plugin" that matches its shape. This is the same idea.